specs/beacon-chain.md

```diff
@@ -746,7 +746,7 @@ For each `SpecialRecord` `obj` in `active_state.pending_specials`:
 
 * **[covers logouts]**: If `obj.kind == LOGOUT`, interpret `data[0]` as a validator index as an `uint32` and `data[1]` as a signature. If `BLSVerify(pubkey=validators[data[0]].pubkey, msg=hash(LOGOUT_MESSAGE + bytes8(fork_version)), sig=data[1])`, where `fork_version = pre_fork_version if slot < fork_slot_number else post_fork_version`, and `validators[i].status == ACTIVE`, run `exit_validator(data[0], crystallized_state, penalize=False, current_slot=block.slot)`
 * **[covers `NO_DBL_VOTE`, `NO_SURROUND`, `NO_DBL_PROPOSE` slashing conditions]:** If `obj.kind == CASPER_SLASHING`, interpret `data[0]` as a list of concatenated `uint32` values where each value represents an index into `validators`, `data[1]` as the data being signed and `data[2]` as an aggregate signature. Interpret `data[3:6]` similarly. Verify that both signatures are valid, that the two signatures are signing distinct data, and that they are either signing the same slot number, or that one surrounds the other (ie. `source1 < source2 < target2 < target1`). Let `indices` be the list of indices in both signatures; verify that its length is at least 1. For each validator index `v` in `indices`, if its `status` does not equal `PENALIZED`, then run `exit_validator(v, crystallized_state, penalize=True, current_slot=block.slot)`
-* **[covers RANDAO updates]**: If `obj.kind == RANDAO_REVEAL`, interpret `data[0]` as an integer and `data[1]` as a hash32. Set `validators[data[0]].randao_commitment = data[1]`.
+* **[covers RANDAO updates]**: If `obj.kind == RANDAO_CHANGE`, interpret `data[0]` as a `uint64`, `data[1]` as a hash32, and `data[3]` as a `uint64`. Set `validators[data[0]].randao_commitment = data[1]`, and set `validators[data[0]].randao_last_change = data[2]`.
```

Comment:
Since the first & third elements are `uint64` (8 bytes), it should be:
interpret `data[0:8]` as a `uint64`, `data[8]` as a `hash32`, and `data[9:8]` as a `uint64`.

Reply:
I think we can hint the meaning of the fields more explicitly for readability:

* **[covers RANDAO updates]**: If `obj.kind == RANDAO_CHANGE`, interpret `data[0:8]` as a `uint64`, `data[8]` as a `hash32`, and `data[9:8]` as a `uint64`, where `block_proposer_index = data[0:8]`, `randao_commitment  = data[8]`, and `randao_last_change 
 = data[9:8]`. Set `validators[block_proposer_index].randao_commitment = randao_commitment`, and set `validators[block_proposer_index].randao_last_change = randao_last_change`.